test/e2e/gateway/003_invalid_forward_to_test.go

```diff
@@ -0,0 +1,165 @@
+// Copyright Project Contour Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build e2e
+
+package gateway
+
+import (
+	"context"
+	"testing"
+
+	"github.com/projectcontour/contour/test/e2e"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	gatewayv1alpha1 "sigs.k8s.io/gateway-api/apis/v1alpha1"
+)
+
+func testInvalidForwardTo(t *testing.T, fx *e2e.Framework) {
+	namespace := "gateway-003-invalid-forward-to"
+
+	fx.CreateNamespace(namespace)
```

Comment:
Yeah, would definitely like to get there. For the current set of tests it's pretty easy because generally most things are encapsulated in a namespace so deleting the namespace cleans it all up and is easy, but doing full tracking would be better. I've already run into some challenges with the Gateway tests given that the Gateway has to live in a different NS.